ui: display more load stats on /hotranges page

kvoli opened the pull request:
First two commits are from https://github.com/cockroachdb/cockroach/pull/95388.

Previously the hot ranges page only showed one statistic that helped
identify high load ("hot") ranges: queries per second (QPS). QPS is a
measure of the number of batch request a replica processed per second,
averaged over the last 30 minutes. If the batch request composition is
non-uniform in terms of incidental load on the cluster, the correlation
between QPS and importance of a range to the end user weakens.

This commit adds more statistics per-range to the hot ranges ui page to
provide better insight into load composition. The statistics are rated
(per-second) and averaged over the last 30 minutes, the same as the
existing load statistic QPS.

- CPU: CPU time used in processing this range.
- Write (keys):  number of keys written on this range.
- Write (bytes): number of bytes written on this range.
- Read (keys):   number of keys read on this range.
- Read (bytes):  number of bytes read on this range.

The ranges that are displayed on the hot ranges page can be ordered
using these statistics as a key. An example page view is shown below
running kv50.

![image](https://user-images.githubusercontent.com/39606633/213811279-07348a81-05dd-463f-a970-706083c681c3.png)

depends on: https://github.com/cockroachdb/cockroach/pull/95388

resolves: https://github.com/cockroachdb/cockroach/issues/95386

Release note (ui change) Add write bytes, write keys, read bytes, read
keys and cpu statistics to the `/hotranges` db-console page. These
statistics are the rated average over the last 30 minutes.

kvoli committed:
ui: display more load stats on /hotranges page

Previously the hot ranges page only showed one statistic that helped
identifying high load ("hot") ranges: queries per second (QPS). QPS is a
measure of the number of batch request a replica processed per second,
averaged over the last 30 minutes. If the batch request
composition is non-uniform in terms of incidental load on the
cluster, the correlation between QPS and importance of a range
to the end user weakens.

This commit adds more statistics per-range to the hot ranges ui page to
provide better insight into load composition. The statistics are rated
(per-second) and averaged over the last 30 minutes, the same as the
existing load statistic QPS.

- CPU:           CPU time used in processing this range.
- Write (keys):  number of keys written on this range.
- Write (bytes): number of bytes written on this range.
- Read (keys):   number of keys read on this range.
- Read (bytes):  number of bytes read on this range.

The ranges that are displayed on the hot ranges page can be ordered
using these statistics as a key.

resolves: #95386

Release note (ui change) Add write bytes, write keys, read bytes, read
keys and cpu statistics to the `/hotranges` db-console page. These
statistics are the rated average over the last 30 minutes.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/95190)
<!-- Reviewable:end -->

kvoli committed:
kvserver,server: add range cpu time hotrange api

Various range load statistics such as read(write) bytes(keys) exist on
the hot ranges (v1) and range statistics api. This commit adds the
additional statistic, `CPUTimePerSecond`. This statistic is the recent
average cpu usage per second of the range in nanoseconds.

resolves: #95384

Release note: None

kvoli committed:
server: add more load stats to hot ranges v2 api

Previously, only QPS was included in the hot ranges v2 api. This commit
adds the following range statistics to the API:

```
WritesPerSecond
ReadsPerSecond
WriteBytesPerSecond
ReadBytesPerSecond
CPUTimePerSecond
```

See the associated comments in `status.proto` for a description of each
load statistic.

These may be used to determine the hottest ranges. Note however that
these are sorted by default by QPS. The ordering of ranges w.r.t the new
load statistics has no guarantees. Likewise, ranges may be omitted which
have high relative values for the new load statistics but relatively low
QPS. In practice however, there usually exists a strong correlation that
ensures that such ranges would not be omitted entirely.

resolves: #95383

Release note: None

maryliag commented:
A drive by comment... 
on the SQL Activity pages we had separate column for read and writes, but that was creating a lot of columns and consequently more horizontal scrolling. Our solution was to group and have a column called `Rows Processed` and the values are `1 k Reads / 0 Writes` for example. 
You could change to something similar to keep the pages aligned, so `Keys Processed` and `Bytes Processed`.

kvoli commented:
> Our solution was to group and have a column called `Rows Processed` and the values are `1 k Reads / 0 Writes` for example. 
> You could change to something similar to keep the pages aligned, so `Keys Processed` and `Bytes Processed`.

I agree about it being a bit wide. A requirement is being able to sort each dimension independently. Often hot ranges is used to debug and tune load based splitting and load based rebalancing.

I've seen a few support cases where one range deceptively has quite low QPS but extreme skew between read/writes. Being able to sort independently alleviates mental effort here.

I'm not a very adept contributor to the db-console  area. I noticed other tables had a show column feature, that could be the right solution? I tried adding here but failed once I realized I couldn't easily transfer the structure. Image below.

![image-1.png](https://user-images.githubusercontent.com/39606633/213838677-be28fd32-c787-4d01-a1d7-8d92ce8c2e4b.png)

koorosh reviewed (approved):
Front-end part <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> and ready to merge after rebasing on top of #95388

> Reviewed 4 of 4 files at r1, 6 of 6 files at r2, 6 of 6 files at r3, 1 of 4 files at r7.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95190#-:-NM_20lA43TXxClsI-8w:b72ifne)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @kvoli)



<!-- Sent from Reviewable.io -->

kvoli commented:
TYFTR I'm going to merge this and close https://github.com/cockroachdb/cockroach/pull/95388

bors r=koorosh
